viewing_party/party.py

```diff
@@ -1,23 +1,183 @@
+        return avg_ratings
+    return 0.0
 
+def get_most_watched_genre(user_data):
+    '''
+    takes in user_data, returns the most common genre in watched
+    '''
+    genres = []
+    genre_most_count = 0
+    most_common_genre = None
+
+    for movie in user_data["watched"]:
+        genres.append(movie["genre"])
+        for genre in genres:
+            if genres.count(genre) > genre_most_count:
+                genre_most_count = genres.count(genre)
+                most_common_genre = genre    
+    
+    return most_common_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
+# # -----------------------------------------
+
+def get_common_watched(user_data):
+    '''
+    takes in user_data
+    returns the movies that both the user and friends have watched in a list
+    '''
+    friends_data = user_data["friends"]
+    common_watched = []
+
+    for movie in user_data["watched"]:
+        for friend in friends_data:
+            for friend_movie in friend["watched"]:
+                if movie["title"] == friend_movie["title"]:
+                    common_watched.append(movie)
+    return common_watched
+
+def get_unique_watched(user_data):
+    '''
+    takes in user_data, 
+    returns a list of dictionaries for each movie that only the user has watched
+    '''
+    # friends data is a list of dictionaries with 
+    # duplicate_watched = [title for movie in user_data["watched"] for title in movie]
+    unique_watched = []
+    common_watched = get_common_watched(user_data)
+
+    for movie in user_data["watched"]:
+        if movie not in common_watched and movie not in unique_watched:
+            unique_watched.append(movie)
+
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    '''
+    takes in user_data, 
+    returns a list of dictionaries for each movie that only the friends have watched
+    '''
+    friends_data = user_data["friends"]
+    common_watched = get_common_watched(user_data)
+    unique_watched = []
+    
+    for friend in friends_data:
+        for movie in friend["watched"]:
+            if movie not in common_watched and movie not in unique_watched:
+                unique_watched.append(movie)
+    return unique_watched
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    '''
+    takes in user_data, 
+    returns a list of dictionaries for each movie that the 
+    user has not watched from friends_unique_watched 
+    and are hosted within users subscription
+    '''
+    user_subscriptions = user_data["subscriptions"]
+    friends_movies = get_friends_unique_watched(user_data)
+    movie_recs = []
+    
+    for movie in friends_movies: 
+        if movie["host"] in user_subscriptions:
+            movie_recs.append(movie)

```

Comment:
Really like the use of variables on line 140-141 to extract and name the data structures we'll be working with.